spec/unit/lcm/lcm2_spec.rb

```diff
@@ -45,10 +37,125 @@
       allow(domain).to receive(:data_products)
     end
 
-    it 'skips actions in skip_actions' do
-      expect(GoodData::LCM2::CollectSegments).not_to receive(:call)
-      expect(GoodData::LCM2::SynchronizeUsers).not_to receive(:call)
-      GoodData::LCM2.perform('users', params)
+    context 'when skip_actions specified' do
+      let(:params) do
+        params = {
+          skip_actions: %w(CollectSegments SynchronizeUsers),
+          GDC_GD_CLIENT: client,
+          GDC_LOGGER: logger,
+          domain: domain
+        }
+        GoodData::LCM2.convert_to_smart_hash(params)
+      end
+
+      it 'skips actions in skip_actions' do
+        expect(GoodData::LCM2::CollectSegments).not_to receive(:call)
+        expect(GoodData::LCM2::SynchronizeUsers).not_to receive(:call)
+        GoodData::LCM2.perform('users', params)
+      end
+    end
+
+    context 'when domain is not specified in params' do
+      let(:params) do
+        params = {
+          GDC_GD_CLIENT: client,
+          GDC_LOGGER: logger
+        }
+        expect { raise RuntimeError }.to raise_error
+        GoodData::LCM2.convert_to_smart_hash(params)
+      end
+
+      it 'exception should be thrown' do
```

Comment:
warning: Similar code found in lcm2_spec.rb:68, lcm2_spec.rb:103 (mass = 40) (flay)